doctr/models/recognition/core.py

```diff
@@ -93,8 +95,29 @@ def __call__(
             if any(crop.ndim != 3 for crop in crops):
                 raise ValueError("incorrect input shape: all crops are expected to be multi-channel 2D images.")
 
+            splitted_crops: List[np.ndarray] = []
+            splitted_idxs: List[List[int]] = []
+            for crop in crops:
+                h, w = crop.shape[:2]
+                aspect_ratio = w / h
+                if aspect_ratio > self.critical_ar:
+                    # Determine the number of crops, reference aspect ratio = 4 = 128 / 32
+                    n_crops = int(aspect_ratio // 4)
+                    # Find the new widths, additional dilation factor to overlap crops
+                    width = int(self.dil_factor * w / n_crops)
```

Comment:
Better to avoid casting it to int right now (will be done on line 113)
This could lead to rounding errors otherwise